server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3005,20 +3027,75 @@ private static ActionListener<Void> fileQueueListener(
         });
     }
 
-    private static InputStream maybeRateLimit(InputStream stream, Supplier<RateLimiter> rateLimiterSupplier, CounterMetric metric) {
-        return new RateLimitingInputStream(stream, rateLimiterSupplier, metric::inc);
+    private static void mayBeLogRateLimits(BlobStoreTransferContext context, RateLimiter rateLimiter, long time) {
+        logger.info(
+            () -> new ParameterizedMessage(
+                "Rate limited blob store transfer, context [{}], for duration [{} ms] for configured rate [{} MBps]",
+                context,
+                TimeValue.timeValueNanos(time).millis(),
+                rateLimiter.getMBPerSec()
+            )
+        );
+    }
+
+    private static InputStream maybeRateLimit(
+        InputStream stream,
+        Supplier<RateLimiter> rateLimiterSupplier,
+        CounterMetric metric,
+        BlobStoreTransferContext context
+    ) {
+        return new RateLimitingInputStream(stream, rateLimiterSupplier, (t) -> {
+            mayBeLogRateLimits(context, rateLimiterSupplier.get(), t);
+            metric.inc(t);
+        });
+    }
+
+    private static OffsetRangeInputStream maybeRateLimitRemoteTransfers(
+        OffsetRangeInputStream offsetRangeInputStream,
+        Supplier<RateLimiter> rateLimiterSupplier,
+        CounterMetric metric,
+        BlobStoreTransferContext context
+    ) {
+        return new RateLimitingOffsetRangeInputStream(offsetRangeInputStream, rateLimiterSupplier, (t) -> {
+            mayBeLogRateLimits(context, rateLimiterSupplier.get(), t);
+            metric.inc(t);
+        });
     }
 
     public InputStream maybeRateLimitRestores(InputStream stream) {
         return maybeRateLimit(
-            maybeRateLimit(stream, () -> restoreRateLimiter, restoreRateLimitingTimeInNanos),
+            maybeRateLimit(stream, () -> restoreRateLimiter, restoreRateLimitingTimeInNanos, BlobStoreTransferContext.SNAPSHOT_RESTORE),
```

Comment:
Why are we rate limiting an input stream twice here? Is it because we want the snapshot/restore to use the common bandwidth that is shared across remote store as well and throttling should work regardless?